### DISCUSSION TITLE
How to use skywalking in react-native

### DISCUSSION THREAD

**Discussion by Yshanchui** [apache/skywalking#10290] [Q&A] [answered]
Skywalking-java, skywalking-python, skywalking-client-js have been used in the organization. Would like to ask how to use skywalking in the app developed by react-native.

**Comment by wu-sheng** [comment, marked as answer]
What do you mean `how`? Could you describe more clear?

**Reply by Yshanchui** [reply]
In the app developed by react-native, I tried to use skywalking-client-js as link tracking, and found that the location and window.location used in skywalking-client-js cannot be used in react-native, resulting in failure Use. At the same time, I also tried skywalking-nodejs. On Android phones, it will prompt that the OS cannot recognize this type of error. So I would like to ask whether the app developed under react-native can use skywalking on the agent side.

**Reply by Superskyyy** [reply]
@Yshanchui I believe client-js was not tested for running in react-native yet. If you'd like to help test and contributing the support would be appreciated.

**Reply by Fine0830** [reply]
Currently, there is no agent to support this app.

**Reply by wu-sheng** [reply]
I think for `react-native` stack, most likely we need a totally new agent. 

**Written in JavaScript—rendered with native code** <- This kind of statement usually means, the agent has to be involved in compiling stage, rather than most agents today working in the runtime.

It would be nice to see the community could move forward to the mobile scope for tracing and metrics/logs collecting.

**Reply by Fine0830** [reply]
Agree. Client-JS is used for web applications, we need a new agent for mobile application.

**Reply by wu-sheng** [reply]
We could consider planning this later in 2023. After all, I can see the v9 booster UI is moving forward stable :)

**Reply by Fine0830** [reply]
Great.

**Comment by wu-sheng** [comment]
I start a survey about mobile stack monitoring, let's see.

https://github.com/apache/skywalking/discussions/10292
